contracts/Loans.sol

```diff
@@ -247,7 +247,7 @@ contract Loans is DSMath {
 	function push(bytes32 loan) public { // Fund Loan
 		require(sechs[loan].set);
     	require(bools[loan].pushed == false);
-    	tokes[loan].transferFrom(msg.sender, address(this), prin(loan));
+    	require(tokes[loan].transferFrom(msg.sender, address(this), prin(loan)));
```

Comment:
In `Loans.push`, a failed `transferFrom` could mean that the loan appears to be funded when it is actually not. An attacker could use this to steal that ERC20 token from other users. The attacker would create a loan, supply insufficient funds such that the transfer failed, and then cancel the loan. This would cause the `Loans` contract to send any tokens it was holding to the attacker: